### PR TITLE
Added unix-like commands to FileUtils

### PR DESCRIPTION
**Pull request by ghivert**
Hi.
I tried to add some unix-like commands in FileUtils, as asked in request #1058.
Hope it will be good.

### CODE REVIEW

**Comment by asterite** [comment]
@ghivert Awesome!! ❤️ 

I'll review it soon.

File: src/file_utils.cr [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+  end
+
+  # Changes the current working firectory of the process to the given string *path*
+  # and invoked the block, restoring the original working directory when the block exits.
+  # Alias of Dir.cd with block.
+  # ```
+  # FileUtils.cd("to/directory") { puts "Do something useful here!" }
+  # ```
+  def cd(path : String, &block)
+    Dir.cd(path, &block)
```
Comment (asterite) [inline src/file_utils.cr]: Please implement this as:

```cr
Dir.cd(path) { yield }
```

This avoids closures, as described [here](https://crystal-lang.org/docs/syntax_and_semantics/block_forwarding.html)  (might be improved in the future)

**Comment by RX14** [comment]
@asterite Hopefully not too off-topic, but why exactly do we have `FileUtils` and methods on `File:Class`? What's the split?

**Comment by asterite** [comment]
@ghivert Looks really good! :-)

The only thing I would change (in addition to the block comment above) is to mark methods that you have right now returning `0` as returning `nil`. You can do this by placing ` : Nil` after the method signature, and removing the `0` at the end. There's no need to put a `nil` at the end. This will automatically make the method return `nil`, and it also shows on the docs so users know that the method doesn't return anything useful.

**Comment by asterite** [comment]
@RX14 The idea is that `FileUtils` is a bit more convenient that both `File` and `Dir`, combines functions from both of them, accept enumerables in many places where in `File` and `Dir` just a single argument is accepted, and have unix-like names. You can do:

```cr
include FileUtils

# Now you can write things almost as in a unix shell
cd "foo"
mv "bar", "baz"
mkdir_p "qux"
```

In this particular case these aliases don't bother because it's not like there will be multiple possible implementations of this and these aliases will need to be defined over and over.

**Comment by ghivert** [comment]
@asterite Actually, functions returned 0 to mimic shell behavior. But it seems better for me now. :)

**Comment by asterite** [comment]
@ghivert Sure, that would be great!

I forgot about this PR, it looks good now. Thank you so much for this! ❤️

**Pull request merged**